onnx/defs/schema.cc

```diff
@@ -829,63 +831,52 @@ const FunctionProto* OpSchema::GetFunction() const {
   return GetFunctionWithOpsetVersion(since_version_);
 }
 
-const FunctionProto*
-OpSchema::GetFunctionWithOpsetVersion(int opset_version, const std::string& domain, bool fail_on_invalid_op) const {
-  return const_cast<OpSchema*>(this)->GetFunctionWithOpsetInternal(opset_version, domain, fail_on_invalid_op);
+const FunctionProto* OpSchema::GetFunctionWithOpsetVersion(int requested_opset_version) const {
+  return const_cast<OpSchema*>(this)->GetFunctionWithOpsetInternal(requested_opset_version);
 }
 
-void OpSchema::ValidateReferencedOpsInFunciton(
+// when requesting a function at loading time,
+// requested_opset_version does not have to be the same as function_since_version.
+// When they are not the same, it is necessary to verify that ops used to define the function
+// are not updated between function_since_version and requested_opset_version (include requested_opset_version).
+// this call only validate ops in the default domain.
+// TODO: validate ops in other domains.
+bool OpSchema::ValidateReferencedOpsInFuncton(
     const FunctionProto* function,
     int requested_opset_version,
     int function_since_version,
-    const std::string& domain,
-    bool fail_on_invalid_op) {
+    std::set<std::string>* updated_ops) const {
```

Comment:
Should we use a reference to indicate that this input should be non-null?

Reply:
It does allow nullptr in which case the caller is not interested in the specific ops that are updated after since_version.